packages/core/src/enqueue_links/enqueue_links.ts

```diff
@@ -294,6 +294,20 @@ export function resolveBaseUrlForEnqueueLinksFiltering({
         return finalUrlOrigin;
     }
 
+    // If the user wants to ensure the same domain is accessed, regardless of subdomains, we check to ensure the domains match
+    // Returning undefined here is intentional! If the domains don't match, having no baseUrl in enqueueLinks will cause it to not enqueue anything
+    // which is the intended behavior (since we went off domain)
+    if (enqueueStrategy === EnqueueStrategy.SameDomain) {
+        const originalHostname = getDomain(originalUrlOrigin, { mixedInputs: false })!;
+        const finalHostname = getDomain(finalUrlOrigin, { mixedInputs: false })!;
+
+        if (originalHostname === finalHostname) {
+            return finalUrlOrigin;
+        }
+
+        return undefined;
+    }
```

Comment:
Not sure if this is a problem or not, so just leaving it here for consideration. In the Crawlee tutorials which i'm now rewriting, I saw an `origin` comparison used instead of `hostname`. The thing is, `origin` includes the protocol and the hostname does not. This means that URLs with the same hostname, but different protocol (eg http vs https), would be filtered out.

Please check or confirm that it behaves correctly in Crawlee and it was just wrong in the example code.